gitenberg/book.py

```diff
@@ -182,6 +182,7 @@ def repo(self):
     def all(self):
         try:
             self.fetch()
+            self.local_repo = LocalRepo(self.local_path)
```

Comment:
I'm worried about this step - will look at it in the morning. the local_repo gets executed in other places; the flow is a bit of a mess. (of course the correct solution is to have tests for each path, but....)

Reply:
Idea: if you're not sure after looking again, it seems like wrapping this in `if not self.local_repo` would be pretty safe. I can't think of any way changing it with that addition would break any currently-functional workflows, because if it's not set then the next line fails anyways. What do you think?

Reply:
it looks like the problem is that the first line in make(), which gets called in the next line is
`local_repo = LocalRepo(self)`
which doesn't even have the right argument, so maybe instead
`self.local_repo = LocalRepo(self.local_path)`

Reply:
p.s. good example of why sleep is the best debugger

Reply:
I don't see that... The only places I see `LocalRepo(` outside of tests are:
```console
$ git grep 'LocalRepo(' | fgrep -v tests
gitenberg/book.py:            self.local_repo = LocalRepo(self.local_path)
gitenberg/book.py:            self.local_repo = LocalRepo(self.local_path)
gitenberg/clone.py:    return LocalRepo(None, cloned_repo=vat.local_repo)
gitenberg/local_repo.py:class LocalRepo(object):
notebooks/covers.ipynb:    "lr = LocalRepo('/Documents/gitenberg/library/51118')\n",
```
The first one in `book.py` is in `__init__`, and the second is the one I added. The one in clone doesn't apply when creating new repos. Maybe you have a local almost-fix for this problem that isn't in master?

Reply:
it seems I was looking at an old version

Reply:
https://github.com/gitenberg-dev/gitberg/blob/b815b46d6c7e52d368dd3103dcce1914f510c6a3/gitenberg/book.py

Reply:
This is coming back to me. The complexity is that the local repo might be
- cloned from github (folder name is a repo_name)
- created by rsync (folder name is a number) not yet pushed to github
- created by rsync but previously pushed to github
all() is supposed to be used for books not yet in gitenberg, but nothing prevents it from being called when there's already a repo.

All in all, I think your idea about wrapping it in a test is the right thing to do

Reply:
Sounds good, and done.